src/test/java/org/jenkinsci/plugins/docker/commons/SampleDockerBuilder.java

```diff
@@ -39,9 +39,9 @@ public DockerRegistryEndpoint getRegistry() {
     @Override
     public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
         // prepare the credentials to talk to this docker and make it available for docker you'll be forking
-        KeyMaterial key = server.materialize(build);
+        KeyMaterialFactory keyMaterialFactory = server.newKeyMaterialFactory(build).plus(registry.newKeyMaterialFactory(build));
+        KeyMaterial key = keyMaterialFactory.materialize();
         try {
-            key = key.plus(registry.materialize(build));
```

Comment:
Good, this is a little simpler. In the former code, though it might not have been apparent, we were handling the case that `server.materialize` succeeded yet `registry.materialize` failed by `close`ing the material from `server.materialize`.